### PR TITLE
Adding encryption feature for Gateway Mule 4 

### PR DESCRIPTION
**Pull request by fermujica**

### CODE REVIEW

File: modules/ROOT/pages/api-gateway-encryption-mule4.adoc [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+
+In the example above, your `client_id` would look as below:
+
+[source]
+--
+anypoint.platform.client_id=![smzbg+aV2HbiC9Fv8KBsZpD5PlXiplsfhZ42RWNEbHhUeZ9ZIDftljUx6sYpuQ+b]
+--
+
+[NOTE]
+The `![` prefix and  `]` suffixes are required, as they indicate the runtime that the enclosed value is encrypted.
```
Comment (LorenaSoledad) [inline modules/ROOT/pages/api-gateway-encryption-mule4.adoc]: I think it should be `suffix` here.

File: modules/ROOT/pages/api-gateway-encryption-mule4.adoc [hunk 2]
```diff
@@ -0,0 +1,252 @@
+= Gateway Startup Encryption in Mule 4
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+Gateway startup encryption allows you to encrypt sensitive information required to configure the connection between your Mule application and Anypoint Platform.
```
Comment (evangelina) [inline modules/ROOT/pages/api-gateway-encryption-mule4.adoc]: It also encrypts sensitive information stored by the Gateway such as policies and contracts.

File: modules/ROOT/pages/api-gateway-encryption-mule4.adoc [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+
+== How It Works
+
+You can encrypt your Anypoint Platform credentials (client ID, client secret, and your proxy password) and add those encrypted values, along with your encryption key to your runtime's `wrapper.conf` file.
+
+While the runtime is starting it checks for an encryption key. Then it gets the client ID and client secret values from its `wrapper.conf` file. If some of the values in that file are enclosed within the encryption indicators (`![` and `]`), the runtime uses the encryption key it loaded earlier to decrypt the values and use its original content.
+
+This process can also be used to encrypt and decrypt sensitive values in your API policies.
```
Comment (evangelina) [inline modules/ROOT/pages/api-gateway-encryption-mule4.adoc]: Actually, the process described above is not used to encrypt and decrypt policy data. Just by supplying the encryption key the policies and contracts are encrypted, but it's not tied to the client id, secret config.

File: modules/ROOT/pages/api-gateway-encryption-mule4.adoc [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+If you already have all your policies encrypted and then needed to restart the runtime without an encryption key, all policies will be downloaded again, and all the sensitive information won't be encrypted.
+--
+* `anypoint.platform.encryption_mode`. +
+This property can take two values:
++
+[%autowidth.spread,cols="a,a"]
+|===
+| `FULL` | All values are encrypted.
+| `SENSITIVE_ONLY` | Only values that you marked as `sensitive` are encrypted.
+|===
```
Comment (evangelina) [inline modules/ROOT/pages/api-gateway-encryption-mule4.adoc:176]: We should show that the default is full. @LorenaSoledad can you confirm the default value?

File: modules/ROOT/pages/api-gateway-encryption-mule4.adoc [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+[CAUTION]
+--
+If you are changing from `FULL` to `SENSITIVE_ONLY`, any parameter not marked as `sensitive` will be shown as plain text in your filesystem.
+--
+
+== Policies That Support Encryption
+
+Below is a list of all policies that support encryption in their values, along with their minimum required version to support this feature. +
+Earlier versions of these policies, and policies not listed below will not encrypt the policies values even when the gateway capability detects and encryption key in its configuration.
+
```
Comment (evangelina) [inline modules/ROOT/pages/api-gateway-encryption-mule4.adoc:190]: Typo, it should be "even when the gateway capability detects **an** encryption key in its configuration."

File: modules/ROOT/pages/api-gateway-encryption-mule4.adoc (hunk 2 above)
Comment (fermujica) [reply modules/ROOT/pages/api-gateway-encryption-mule4.adoc]: You are right, maybe changing it to something like this:

> Gateway startup encryption allows you to encrypt sensitive information stored by the runtime such as policies, contracts, and the required credentials to configure the connection between your Mule application and Anypoint Platform.

File: modules/ROOT/pages/api-gateway-encryption-mule4.adoc (hunk 5 above)
Comment (fermujica) [reply modules/ROOT/pages/api-gateway-encryption-mule4.adoc:190]: Thanks!  Fixed in 5bb7bff

File: modules/ROOT/pages/api-gateway-encryption-mule4.adoc (hunk 4 above)
Comment (LorenaSoledad) [reply modules/ROOT/pages/api-gateway-encryption-mule4.adoc:176]: No, the default is **Sensitive Only**

**Review by LorenaSoledad — DISMISSED** [review]

File: modules/ROOT/pages/api-gateway-encryption-mule4.adoc (hunk 2 above)
Comment (LorenaSoledad) [reply modules/ROOT/pages/api-gateway-encryption-mule4.adoc]: Seems good to me.

File: modules/ROOT/pages/api-gateway-encryption-mule4.adoc (hunk 3 above)
Comment (fermujica) [reply modules/ROOT/pages/api-gateway-encryption-mule4.adoc]: Moved this section to be part of the credentials encryption: bc6547d

File: modules/ROOT/pages/api-gateway-encryption-mule4.adoc [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+[source]
+--
+anypoint.platform.client_id=![smzbg+aV2HbiC9Fv8KBsZpD5PlXiplsfhZ42RWNEbHhUeZ9ZIDftljUx6sYpuQ+b]
+--
+
+[NOTE]
+The `![` prefix and  `]` suffix are required, as they indicate the runtime that the enclosed value is encrypted.
+
+For the runtime to be able to decrypt your secured values, you need to configure your secret key using the `anypoint.platform.encryption_key` property in the wrapper file:
```
Comment (balbifm) [inline modules/ROOT/pages/api-gateway-encryption-mule4.adoc]: I would not suggest having the encryption_key in the wrapper file. Instead it should be passed as an argument when starting the runtime

File: modules/ROOT/pages/api-gateway-encryption-mule4.adoc [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+
+[source]
+--
+anypoint.platform.encryption_key=MyEncryptionKey1
+--
+
+The runtime uses the encryption key configured in the `anypoint.platform.encryption_key` property to decrypt the properties.
+
+When you provide a key, the runtime displays the following message:
```
Comment (balbifm) [inline modules/ROOT/pages/api-gateway-encryption-mule4.adoc]: These messages do not appear by default. Only if com.mulesoft.mule.runtime.gw.deployment is configured in DEBUG level

File: modules/ROOT/pages/api-gateway-encryption-mule4.adoc [hunk 8]
```diff
@@ -0,0 +1,249 @@
...
+|===
+
+== Policies Encryption
+
+The configuration for your API policies is stored in the `policies` directory, and some policies may contain sensitive information that you want to hide from unauthorized parties, such as the password field for the `Basic Authentication` header. +
+Sensitive values in your provided API policies are marked as such in the API Manager UI for policy configuration. When creating a custom policy, you can define which values are sensitive.
+
+== Configuring Policy Encryption
+
+To enable encryption in policies, you need to provide two system properties in your `wrapper.conf` file:
```
Comment (balbifm) [inline modules/ROOT/pages/api-gateway-encryption-mule4.adoc:159]: Second one is optional, if not present, default value is used

**Review by balbifm — DISMISSED** [review]

File: modules/ROOT/pages/api-gateway-encryption-mule4.adoc (hunk 6 above)
Comment (fermujica) [reply modules/ROOT/pages/api-gateway-encryption-mule4.adoc]: You are right. I changed it: 8677bfb

File: modules/ROOT/pages/api-gateway-encryption-mule4.adoc (hunk 7 above)
Comment (fermujica) [reply modules/ROOT/pages/api-gateway-encryption-mule4.adoc]: Added this as debug instructions: e413dcc

Please verify if this is technically accurate

**Comment by fermujica** [comment]
Just added encryption configuration for Mule 3 gateways: 1da305a

File: modules/ROOT/pages/api-gateway-encryption-mule4.adoc [hunk 9]
```diff
@@ -0,0 +1,270 @@
...
+--
+
+== Passing your Encrypted Client ID and Secret
+
+After encrypting your credentials, you need to configure your encrypted values in your `wrapper.conf` file located inside the `conf` directory in the root folder of your Mule runtime:
+
+[source]
+--
+anypoint.platform.client_id.<n>
+anypoint.platform.client_secet.<n>
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/api-gateway-encryption-mule4.adoc]: Seems this is a typo: anypoint.platform.client_secet.<n>, should be anypoint.platform.client_secret.<n>

File: modules/ROOT/pages/api-gateway-encryption-mule4.adoc [hunk 10]
```diff
@@ -0,0 +1,270 @@
+= Gateway Startup Encryption in Mule 4
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+Gateway startup encryption allows you to encrypt sensitive information stored by the runtime such as policies, contracts, and the required credentials to configure the connection between your Mule application and Anypoint Platform.
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/api-gateway-encryption-mule4.adoc]: According to our style guide, "by the runtime" should be either "by Mule" or "by the Mule runtime engine".
Please check if this change is necessary

File: modules/ROOT/pages/api-gateway-encryption-mule4.adoc [hunk 11]
```diff
@@ -0,0 +1,270 @@
...
+
+In the example above, your `client_id` would look as below:
+
+[source]
+--
+anypoint.platform.client_id.<n>=![smzbg+aV2HbiC9Fv8KBsZpD5PlXiplsfhZ42RWNEbHhUeZ9ZIDftljUx6sYpuQ+b]
+--
+
+[NOTE]
+The `![` prefix and  `]` suffix are required, as they indicate the runtime that the enclosed value is encrypted.
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/api-gateway-encryption-mule4.adoc]: runtime > either 'Mule' or 'Mule runtime engine'

File: modules/ROOT/pages/api-gateway-encryption-mule4.adoc [hunk 12]
```diff
@@ -0,0 +1,270 @@
...
+No encryption key provided. Policies and API contracts won't be encrypted.
+----
+
+If the decryption process fails, the runtime displays an error:
+
+----
+Client ID cannot be decrypted
+----
+
+If your credentials can't be decrypted, you cannot establish a connection with Platform, and won't be able to download policies or contracts.
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/api-gateway-encryption-mule4.adoc]: what is "Platform" here? "Anypoint Platform" ? or may be just use "...with the platform".

Comment (fermujica) [reply modules/ROOT/pages/api-gateway-encryption-mule4.adoc]: Right: This should be Anypoint Platform.

File: modules/ROOT/pages/api-gateway-encryption-mule4.adoc [hunk 13]
```diff
@@ -0,0 +1,270 @@
...
+== Configuring Policy Encryption
+
+To enable encryption in policies, you need to provide two system properties in your `wrapper.conf` file:
+
+* `anypoint.platform.encryption_key.<n>`. +
+The encryption key for your policy.
++
+[CAUTION]
+--
+If you already have all your policies encrypted and then needed to restart the runtime without an encryption key, all policies will be downloaded again, and all the sensitive information won't be encrypted.
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/api-gateway-encryption-mule4.adoc]: This sentence sounds misleading, it is trying to say the following: 
"If you have all your policies encrypted, and then you restart Mule without an encryption key,..." ?

File: modules/ROOT/pages/api-gateway-encryption-mule4.adoc [hunk 14]
```diff
@@ -0,0 +1,270 @@
...
+<secure-properties:config key="${anypoint.platform.encryption_key}" file="${encryptedPropertiesFile}" name="encryptionConfiguration">
+   <secure-properties:encrypt algorithm="AES" mode="CBC"/>
+</secure-properties:config>
+{{/if}}
+--
++
+This feature supports only the AES Algorithm in CBC mode. +
+The gateway capability takes the `anypoint.platform.encryption_key` and `encryptedPropertiesFile` values from the system properties that you set at startup time for the gateway.
+. Define the sensitive values in the policy's YAML file. +
+By default, the values in a custom policy are not sensitive. You can define whether a value is sensitive or not in the YAML file describing your policy you can define which values are sensitive using the boolean element `sensitive`:
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/api-gateway-encryption-mule4.adoc]: "describing  your policy you can define" > I think these are two separate sentences "describing  your policy. You can define..." Please check.

**Review by Cristian-Venticinque — APPROVED** [review]

**Pull request merged**